### ISSUE TITLE
Using an IPv6 iscsi target portal causes mount to fail

### ISSUE THREAD

**Issue by growse** [democratic-csi/democratic-csi#219]
Using democratic-csi from helm chart v0.13.1.

If I configure:

```
driver
  config:
    driver: freenas-iscsi
    iscsi:
      targetPortal: '[2001:123:456::1]:3260'
```

And then request a k8s deployment with a pvc with the appropriate storage class, the pod fails to start with:

```
MountVolume.MountDevice failed for volume "pvc-5168eaae-77c4-4361-91e9-6994283d36c3" : rpc error: code = Internal desc = {"code":1,"stdout":"","stderr":"iscsiadm: false is an invalid session ID or path\n"," ││ timeout":false}                                                        
```

The volume is provisioned and shared properly on Freenas, and an iscsi session is opened to the correct portal on the k8s node. However, on the `csi-driver` log, I can see:

```
csi-driver executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:default-aptcacher-iscsi-claim -p [2001:123:456::1]:3260 -o new
csi-driver executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:default-aptcacher-iscsi-claim -p [2001:123:456::1]:3260 -o update --name node.startup --value manual                           
csi-driver executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:default-aptcacher-iscsi-claim -p [2001:123:456::1]:3260 -l
csi-driver executing iscsi command: iscsiadm -m session
csi-driver executing iscsi command: iscsiadm -m session -r false --rescan 
```

The fault here is the last line - it's trying to rescan a session but passing in a session id of `false`, which isn't valid.

It seems this comes from the logic in `src/utils/iscsi.js#L181` (https://github.com/democratic-csi/democratic-csi/blob/v1.7.2/src/utils/iscsi.js#L181) which looks to discover the session id. For whatever reason, it doesn't find it, so returns `false` rather than a session id, which then gets passed through to `rescanSession` in https://github.com/democratic-csi/democratic-csi/blob/v1.7.2/src/utils/iscsi.js#L491.

I suspect the ` portal == i_session.portal` comparison is fault somehow, but not really sure how to debug to get any further.

**Comment by travisghansen** [comment]
Do you know if the login in actually successful? Can you send me the output of `iscsiadm -m session` from a node that has successfully logged into a ipv6 portal?

**Comment by growse** [comment]
The login is successful and a session appears in `iscsiadm -m session`.

**Comment by travisghansen** [comment]
Can you please send the output of that command?

**Comment by growse** [comment]
```
$ iscsiadm -m session
tcp: [104] 192.168.2.2:3260,1 iqn.2005-10.org.freenas.ctl:default-aptly-iscsi-claim (non-flash)
tcp: [105] 192.168.2.2:3260,1 iqn.2005-10.org.freenas.ctl:default-powerdns-data-iscsi-claim (non-flash)
tcp: [106] 192.168.2.2:3260,1 iqn.2005-10.org.freenas.ctl:default-smtp-iscsi-claim (non-flash)
tcp: [108] 192.168.2.2:3260,1 iqn.2005-10.org.freenas.ctl:helmthings-prometheus-server (non-flash)
tcp: [111] [2001:123:456::1]:3260,1 iqn.2005-10.org.freenas.ctl:default-aptcacher-iscsi-claim (non-flash)
```

**Comment by travisghansen** [comment]
Ok thanks! I’ll see what’s wrong with the parsing logic and get a fix in `next` for you to try.

**Comment by travisghansen** [comment]
Fixed here: https://github.com/democratic-csi/democratic-csi/commit/153bce28331f54b2188df4fc940ec8c03d758875

Images have been built and tagged `next` (it's a mutable tag so make sure you force the nodes to pull).

**Comment by travisghansen** [comment]
Also in `latest` and `v1.7.3` now as well.

**Comment by growse** [comment]
Thanks. Updated to the latest via helm (v1.7.4) and have a new error condition!

Driver logs:

```
zfs-iscsi-democratic-csi-node-wr7hf csi-driver 2022-08-02T17:01:37.228460599Z {"host":"thatserverbot","level":"info","message":"new request - driver: FreeNASSshDriver method: NodeStageVolume call: {\"metadata\":{\"user-agent\":[\"grpc-go/1.40.0\"],\"x-forwarded-host\":[\"/var/lib/kubelet/plugins/org.democratic-csi.iscsi/csi.sock\"]},\"request\":{\"publish_context\":{},\"secrets\":\"redacted\",\"volume_context\":{\"storage.kubernetes.io/csiProvisionerIdentity\":\"1659457029385-8081-org.democratic-csi.iscsi\",\"interface\":\"\",\"iqn\":\"iqn.2005-10.org.freenas.ctl:default-aptcacher-iscsi-claim\",\"lun\":\"0\",\"node_attach_driver\":\"iscsi\",\"portal\":\"[2001:123:456::1]:3260\",\"portals\":\"\",\"provisioner_driver\":\"freenas-iscsi\"},\"volume_id\":\"pvc-d352be8a-c104-4309-96a1-43ff9554836f\",\"staging_target_path\":\"/var/lib/kubelet/plugins/kubernetes.io/csi/org.democratic-csi.iscsi/c474b2a2ff9badf0e700802e3a308090e91ba8846669cc29700b0946fce5cd0d/globalmount\",\"volume_capability\":{\"access_mode\":{\"mode\":\"SINGLE_NODE_MULTI_WRITER\"},\"mount\":{\"mount_flags\":[],\"fs_type\":\"ext4\",\"volume_mount_group\":\"\"},\"access_type\":\"mount\"}},\"cancelled\":false}","service":"democratic-csi","timestamp":"2022-08-02T17:01:37.227Z"}
zfs-iscsi-democratic-csi-node-wr7hf csi-driver 2022-08-02T17:01:37.230156402Z executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:default-aptcacher-iscsi-claim -p [2001:123:456::1]:3260 -o new
zfs-iscsi-democratic-csi-node-wr7hf csi-driver 2022-08-02T17:01:37.245997375Z executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:default-aptcacher-iscsi-claim -p [2001:123:456::1]:3260 -o update --name node.startup --value manual
zfs-iscsi-democratic-csi-node-wr7hf csi-driver 2022-08-02T17:01:37.274362527Z executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:default-aptcacher-iscsi-claim -p [2001:123:456::1]:3260 -l
zfs-iscsi-democratic-csi-node-wr7hf csi-driver 2022-08-02T17:01:37.299479591Z executing iscsi command: iscsiadm -m session
zfs-iscsi-democratic-csi-node-wr7hf csi-driver 2022-08-02T17:01:37.322509807Z executing iscsi command: iscsiadm -m session -r 120 --rescan
zfs-iscsi-democratic-csi-node-wr7hf csi-driver 2022-08-02T17:01:37.343020082Z retry - failed condition, not trying again
zfs-iscsi-democratic-csi-node-wr7hf csi-driver 2022-08-02T17:01:39.343469201Z retry - failed condition, not trying again
zfs-iscsi-democratic-csi-node-wr7hf csi-driver 2022-08-02T17:01:41.343171658Z retry - failed condition, not trying again
zfs-iscsi-democratic-csi-node-wr7hf csi-driver 2022-08-02T17:01:43.345455737Z retry - failed condition, not trying again
```

Kubelet logs:
```
Aug 02 17:02:11 thatserverbot kubelet[2910184]: E0802 17:02:11.371561 2910184 csi_attacher.go:344] kubernetes.io/csi: attacher.MountDevice failed: rpc error: code = Unknown desc = unable to attach any iscsi devices
Aug 02 17:02:11 thatserverbot kubelet[2910184]: E0802 17:02:11.373102 2910184 nestedpendingoperations.go:335] Operation for "{volumeName:kubernetes.io/csi/org.democratic-csi.iscsi^pvc-d352be8a-c104-4309-96a1-43ff9554836f podName: nodeName:}" failed. No retries permitted until 2022-08-02 17:02:13.373056366 +0000 UTC m=+595319.309929533 (durationBeforeRetry 2s). Error: MountVolume.MountDevice failed for volume "pvc-d352be8a-c104-4309-96a1-43ff9554836f" (UniqueName: "kubernetes.io/csi/org.democratic-csi.iscsi^pvc-d352be8a-c104-4309-96a1-43ff9554836f") pod "apt-cacher-7f4d85fdd-2drt4" (UID: "3449d8b7-cac0-4a1b-968f-0a022ccf57f1") : rpc error: code = Unknown desc = unable to attach any iscsi devices
Aug 02 17:02:29 thatserverbot kubelet[2910184]: E0802 17:02:29.995508 2910184 kubelet.go:1709] "Unable to attach or mount volumes for pod; skipping pod" err="unmounted volumes=[aptcacher-volume], unattached volumes=[aptcacher-config-volume kube-api-access-vcmpc aptcacher-volume]: timed out waiting for the condition" pod="default/apt-cacher-7f4d85fdd-2drt4"
Aug 02 17:02:29 thatserverbot kubelet[2910184]: E0802 17:02:29.995567 2910184 pod_workers.go:951] "Error syncing pod, skipping" err="unmounted volumes=[aptcacher-volume], unattached volumes=[aptcacher-config-volume kube-api-access-vcmpc aptcacher-volume]: timed out waiting for the condition" pod="default/apt-cacher-7f4d85fdd-2drt4" podUID=3449d8b7-cac0-4a1b-968f-0a022ccf57f1
```

```
$ iscsiadm -m node|grep aptcacher
[2001:123:456::1]:3260,1 iqn.2005-10.org.freenas.ctl:default-aptcacher-iscsi-claim

$ iscsiadm -m session | grep aptcacher
tcp: [120] [2001:123:456::1]:3260,1 iqn.2005-10.org.freenas.ctl:default-aptcacher-iscsi-claim (non-flash)
```

Related: https://github.com/kubernetes/kubernetes/issues/110687 ?

**Comment by travisghansen** [comment]
Funny, I just have a fix for that! I haven't snapped it as a new release yet though but it is in `next` if you want to give it a try: https://github.com/democratic-csi/democratic-csi/commit/516701bd29a41f4b6ffe0b90d26e286c5196706c

**Comment by travisghansen** [comment]
And yes, I was discussing that very bug earlier today with the `csi` team which is what made me realize you would have issues. That specific code is not relevant directly in the case of this driver, but indeed my logic suffers from the exact same issue (and has been resolved in the exact same fashion). I should have `v1.7.5` snapped shortly.

**Comment by travisghansen** [comment]
Fix should be in `latest`, `v1.7.5` and `next` now.

**Comment by growse** [comment]
Success! Many thanks.
